src/components/PopupWithForm.js

```diff
@@ -29,4 +30,8 @@ export default class PopupWithForm extends Popup {
     this._form.reset();
     super.close();
```

Comment:
Сlose modal in the case of successful api response in `then` block in `index.js`, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.